src/mlpack/methods/ann/models/yolov3/yolov3_layer_impl.hpp

```diff
@@ -0,0 +1,265 @@
+    predictionsPerCell * grid, batchSize);
+
+  // Input dimensions: gridSize
+  MatType offset = arma::regspace<MatType>(0, gridSize - 1);
+
+#if ARMA_VERSION_MAJOR < 15
+  // If arma::repcube is not available
+  CubeType anchorsWBS(anchorsW.n_rows, anchorsW.n_cols, batchSize);
+  CubeType anchorsHBS(anchorsH.n_rows, anchorsH.n_cols, batchSize);
+  CubeType xOffset(grid, predictionsPerCell, batchSize);
+
+  arma::Col<ElemType> offsetT =
+    arma::vectorise(arma::repmat(offset.t(), gridSize, 1));
+  CubeType yOffset(grid, predictionsPerCell, batchSize);
+  for (size_t i = 0; i < batchSize; i++)
+  {
+    anchorsWBS.slice(i) = anchorsW;
+    anchorsHBS.slice(i) = anchorsH;
+    xOffset.slice(i) = arma::repmat(offset, gridSize, predictionsPerCell);
+    yOffset.slice(i) = arma::repmat(offsetT, 1, predictionsPerCell);
+  }
+#else
+  CubeType anchorsWBS = arma::repcube(anchorsW, 1, 1, batchSize);
+  CubeType anchorsHBS = arma::repcube(anchorsH, 1, 1, batchSize);
+  CubeType xOffset = arma::repcube(offset, gridSize,
+    predictionsPerCell, batchSize);
+
+  CubeType yOffset = arma::repcube(
+    arma::vectorise(arma::repmat(offset.t(), gridSize, 1)),
+    1, predictionsPerCell, batchSize);
+#endif
+
+  // TODO: add if (this->training). Add check for different batchSize.
+  const size_t cols = predictionsPerCell - 1;
+
+  // x1
+  outputCube.tube(0, 0, grid - 1, cols) =
+    (xOffset + 1 / (1 + arma::exp(-inputCube.tube(0, 0, grid - 1, cols))))
+    * stride
+    - anchorsWBS % arma::exp(inputCube.tube(grid * 2, 0, grid * 3 - 1, cols))
+    / 2;
+
+  // y1
+  outputCube.tube(grid, 0, grid * 2 - 1, cols) = (yOffset + 1 /
+    (1 + arma::exp(-inputCube.tube(grid, 0, grid * 2 - 1, cols)))) * stride
+    - anchorsHBS % arma::exp(inputCube.tube(grid * 3, 0, grid * 4 - 1, cols))
+    / 2;
+
+  // x2
+  outputCube.tube(grid * 2, 0, grid * 3 - 1, cols) =
+    (xOffset + 1 / (1 + arma::exp(-inputCube.tube(0, 0, grid - 1, cols))))
+    * stride
+    + anchorsWBS % arma::exp(inputCube.tube(grid * 2, 0, grid * 3 - 1, cols))
+    / 2;
+
+  // y2
+  outputCube.tube(grid * 3, 0, grid * 4 - 1, cols) = (yOffset + 1 /
+    (1 + arma::exp(-inputCube.tube(grid, 0, grid * 2 - 1, cols)))) * stride
+    + anchorsHBS % arma::exp(inputCube.tube(grid * 3, 0, grid * 4 - 1, cols))
+    / 2;
+
+  // apply logistic sigmoid to objectness and classification logits.
+  outputCube.tube(grid * 4, 0, outputCube.n_rows - 1, cols) = 1. /
+    (1 + arma::exp(-inputCube.tube(grid * 4, 0, inputCube.n_rows - 1, cols)));
+
+  // Reshape, for each batch item.
+  for (size_t i = 0; i < reshapedCube.n_slices; i++)
+  {
+    reshapedCube.slice(i) = arma::reshape(
+      arma::reshape(
+        outputCube.slice(i), grid, numAttributes * predictionsPerCell).t(),
+        numAttributes, predictionsPerCell * grid);
+  }
+}
+
+template <typename MatType>
+void YOLOv3Layer<MatType>::Backward(
+    const MatType& input,
+    const MatType& output,
+    const MatType& gy,
+    MatType& g)
```

Comment:
```suggestion
    const MatType& /* input */,
    const MatType& /* output */,
    const MatType& /* gy */,
    MatType& /* g */)
```
Just to fix compilation warnings?